spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -135,23 +139,20 @@ private static void buildFrom(SpannerPersistentEntity<?> persistentEntity,
 		stringBuilder.append("FROM " + persistentEntity.tableName() + " ");
 	}
 
-	public static void buildOrderBy(SpannerPersistentEntity<?> persistentEntity,
-			StringBuilder stringBuilder, Sort sort) {
-		if (sort.isSorted()) {
-			stringBuilder.append("ORDER BY ");
-			StringJoiner orderStrings = new StringJoiner(" , ");
-			sort.iterator().forEachRemaining(o -> orderStrings.add(ordering(persistentEntity, o)));
-			stringBuilder.append(orderStrings.toString());
+	public static StringBuilder applySort(Sort sort, StringBuilder sql,
+			Function<Order, String> sortedPropertyNameFunction) {
+		if (sort == null || sort.isUnsorted()) {
+			return sql;
 		}
-	}
-
-	private static String ordering(SpannerPersistentEntity<?> persistentEntity, Order order) {
-		return persistentEntity.getPersistentProperty(
-				order.getProperty()).getColumnName() + " " + toString(order);
-	}
-
-	private static String toString(Order order) {
-		return order.isAscending() ? "ASC" : "DESC";
+		sql.append(" ORDER BY ");
+		StringJoiner sj = new StringJoiner(" , ");
+		sort.iterator().forEachRemaining(o -> {
+			String sortedPropertyName = sortedPropertyNameFunction.apply(o);
+			String sortedProperty = o.isIgnoreCase() ? "LOWER(" + sortedPropertyName + ")"
```

Comment:
What if the `sortedPropertyName` not a String, but a something like a number? Wouldn't passing it to `LOWER()` cause a failure?

Reply:
OK, got it.